test/alternator/test_provisioned_throughput.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Test for ProvisionedThroughput
+# ProvisionedThroughput is part of a table definition
+# The following tests make sure we can get, set and update its value
+
+import pytest
+from botocore.exceptions import ClientError
+from decimal import Decimal
+from test.alternator.util import random_string, random_bytes, new_test_table
+
+# When creating a table with PROVISIONED billing mode, ProvisionedThroughput must be explicitly set,
+# and the same values should be reflected when the table is described.
+# This test ensures that when a table is created with the specified billing mode and ProvisionedThroughput values,
+# the `describeTable` operation returns the correct ProvisionedThroughput values.
+def test_create_table(dynamodb):
+    KeySchema=[ { 'AttributeName': 'p123', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c4567', 'KeyType': 'RANGE' }
+        ]
+    AttributeDefinitions=[
+                { 'AttributeName': 'p123', 'AttributeType': 'S' },
+                { 'AttributeName': 'c4567', 'AttributeType': 'S' },
+    ]
+
+    ProvisionedThroughput={
+        'ReadCapacityUnits': 6,
+        'WriteCapacityUnits': 7
+    }
+    with new_test_table(dynamodb,
+                        KeySchema=KeySchema,
+        AttributeDefinitions=AttributeDefinitions,
+        BillingMode='PROVISIONED',
+        ProvisionedThroughput=ProvisionedThroughput) as table:
+        got = table.meta.client.describe_table(TableName=table.name)['Table']
+        if 'BillingMode' in got:
+            # PROVISIONED BillingMode can be omitted, only check if it's present
+            assert got['BillingMode'] == 'PROVISIONED'
```

Comment:
Are you suggesting here (I asked before but I don't remember the answer) that Alternator returns BillingMode=PROVISIONED but DynamoDB doesn't bother to print it because it's the default?
If that's the case, I wonder why we didn't follow DynamoDB on this - but in any case if you write this in the comment, I won't have to ask again when I don't remember :-)

Reply:
I changed the comment, but maybe it's still not clear; DynamoDB doesn't add BillingMode if it's PROVISIONED as it's the default; it is marked as optional in the API documentation. I made sure the test supports both options, and I decided to always return it, I think it's clearer.